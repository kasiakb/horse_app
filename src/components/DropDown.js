import React, { Component } from 'react';


class DropDown extends React.Component {
  options(dropDownInput) {
    return dropDownInput.map((weight) => {
      return <option>{weight}</option>
    })
  }

  render() {
    return (
      <div className='wagaKonia'>
        <form>
          <select>
            {this.options(this.props.dropDownInput)}
          </select>
        </form>,
      </div>
    );
  }
}

export default DropDown;