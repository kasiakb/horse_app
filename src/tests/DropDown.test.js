import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import DropDown from '../components/DropDown';
// import { options } from '../components/DropDown'

describe('DropDown component renders the dropdown with the weight correctly', () => {
  it('renders correctly', () => {
    const dropDownInput=[
      "400kg", 
      "500kg", 
      "600kg", 
      "700kg", 
      "800kg"]
    const onChange=jest.fn()
    const value="500kg"
    
    
    
    
    const rendered = renderer.create(
      <DropDown
        dropDownInput={dropDownInput}
        value={value} 
        onChange={onChange}
      />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});