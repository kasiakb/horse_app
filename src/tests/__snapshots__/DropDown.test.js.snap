// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown component renders the dropdown with the weight correctly renders correctly 1`] = `
<div
  className="wagaKonia"
>
  <form>
    <select
      name="dropDownInput"
      onChange={[MockFunction]}
      type="weight"
      value="500kg"
    >
      <option>
        400kg
      </option>
      <option>
        500kg
      </option>
      <option>
        600kg
      </option>
      <option>
        700kg
      </option>
      <option>
        800kg
      </option>
    </select>
  </form>
</div>
`;
